File: UseCase.js
import { Vocabulary } from "./Vocabulary.js"
import { UseCaseLink } from "./UseCaseLink.js"

export class UseCase {
	constructor(node, graph) {
		this.node = node
		this.graph = graph
	}

	get id() {
		return this.node.id
	}

	get number() {
		return this.id.split("/").pop()
	}

	get title() {
		return [...this.graph.match(this.node, Vocabulary.useCaseTitle)]
			.map(x => x.object)
			.map(x => x.value)
		[0]
	}

	get description() {
		return [...this.graph.match(this.node, Vocabulary.useCaseDescription)]
			.map(x => x.object)
			.map(x => x.value)
		[0]
	}

	get requirements() {
		return [...this.graph.match(undefined, Vocabulary.target, this.node)]
			.map(x => x.subject)
			.map(x => new UseCaseLink(x, this.graph))
	}
}

File: MakeWord/Wrapping/UseCase.cs
using VDS.RDF;
using VDS.RDF.Wrapping;

namespace MakeWord.Wrapping;

internal class UseCase : GraphWrapperNode
{
    protected UseCase(INode node, IGraph graph) : base(node, graph) { }

    internal static UseCase Wrap(INode node, IGraph graph) => new(node, graph);

    internal static UseCase Wrap(GraphWrapperNode node) => Wrap(node, node.Graph);

	internal string Number { get => (this as IUriNode).Uri.ToString().Split("/").Last(); }
	internal string Title { get => this.Singular(Vocabulary.UseCaseTitle, ValueMappings.As<string>); }
	internal string Description { get => this.Singular(Vocabulary.UseCaseDescription, ValueMappings.As<string>); }

	internal ISet<Link> Links { get => this.Subjects(Vocabulary.Target, Link.Wrap, Link.Wrap); }
}
